src/main/scala/cromwell/engine/backend/Backend.scala

```diff
@@ -16,7 +17,7 @@ import scala.util.Try
 
 object Backend {
   lazy val LocalBackend = new LocalBackend
-  lazy val JesBackend = new JesBackend { JesConf } //forces configuration resolution to fail now if something is missing
+  lazy val JesBackend = new JesBackend with JesAuthenticationSingleton { JesConf } //forces configuration resolution to fail now if something is missing
```

Comment:
So we actually need `JesAuthentication` in other places than JesBackend (e.g. `JesEngineFunction`).
The problem I had was that by just having a trait containing a JesConnection, everytime I mixed that trait a new connection was created. This is why I made an object with a JesConnection, and created a trait (`JesAuthenticationSingleton`) referencing this JesConnection so everytime this trait is mixed, it's the same JesConnection that is used.

This still allows to mock entirely `JesAuthentication` in the tests or even later in the code if we find out that we need more than one JesConnection to handle the load.